aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -82,6 +82,32 @@ Tensor truncate(const Tensor& tensor) {
   return tensor;
 }
 
+Tensor& true_divide_out(Tensor& result, const Tensor& self, const Tensor& divisor) {
+  TORCH_CHECK(!isIntegralType(result.scalar_type(), /*includeBool=*/ true),
+            "True division requires a floating output type, but got ",
+            result.scalar_type());
+  auto iter = TensorIterator::binary_op(result, self, divisor, /*check_mem_overlap=*/ true);
+  div_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor true_divide(const Tensor& self, const Tensor& divisor) {
+  if (isIntegralType(self.scalar_type(), /*includeBool=*/ true)
+   && isIntegralType(divisor.scalar_type(), /*includeBool=*/ true)) {
+    const auto scalar_type = typeMetaToScalarType(c10::get_default_dtype());
+    Tensor result = at::empty({0}, self.options().dtype(scalar_type));
+
+    auto iter = TensorIterator::binary_op(result, self, divisor, /*check_mem_overlap=*/ true);
```

Comment:
do we actually need to check_mem_overlap here?  It's an out-of-place op and we should be able to handle the inputs overlapping.